rules/util.js

```diff
@@ -5,30 +5,39 @@ const { version } = require('../package.json');
 
 const REPO_URL = 'https://github.com/jest-community/eslint-plugin-jest';
 
-const expectCase = node =>
-  node.callee.name === 'expect' &&
-  node.arguments.length === 1 &&
-  node.parent &&
-  node.parent.type === 'MemberExpression' &&
-  node.parent.parent;
+const expectCase = node => node.callee.name === 'expect';
```

Comment:
After taking a closer look, I'm a bit worried that removing all the parent checks will break some cases (and some of the new code like `getParentCallExpression` also doesn't do any checks).
Are you sure these are no longer needed, and if so, mind explaining why this can't break?

Reply:
I think the change is fine for several reasons:

1. All the tests are passing. I didn't check all test cases but if we can rely on the tests, the change should be fine.
2. I've looked into all `expectCase` usages, 8 files are using this function in total. And almost in all files, there are further checks after detecting if the node is an `expect` CallExpression. So, intiutively, I can say it looks ok.
3. Actually initial implementation was making sure that expect always gets minimum one argument and expect has a property like `expect(x).property`.  Thus it was missing `expect()` &  `expect(x)` statements. 
  The way I implemented is only checking if it's `expect()`. Rest is up to developers to check further.

Actually after taking another look, now I think we don't even need these checks:
https://github.com/yatki/eslint-plugin-jest/blob/3bafe954fbeb50168116e14a8f350352c806d705/rules/util.js#L12

Because all it does is to make sure there is another property coming after `.not`, `.resolves`, `.rejects` properties. I don't think we need them.

My only concern is during typing the code, linter may complain too early as soon as it detects `expectCase()`. But I couldn't see such case.

I'll try to test the rules manually when I have time tomorrow. Until then, if you notice anything weird, you can just let me know.

If you don't see any other problems, I can update the docs and mark the PR ready for review 🖖🏻

Reply:
Hey, I just ran this on Jest itself. 
There's one error that appears to be correct:
```
jest/packages/expect/src/__tests__/matchers.test.js
  1178:7  error  Async assertions must be awaited or returned  jest/valid-expect
```
Then there's four (really two) errors involving `Promise.all`:
```
jest/packages/jest-mock/src/__tests__/index.test.ts
  589:9  error  Async assertions must be awaited or returned  jest/valid-expect
  590:9  error  Async assertions must be awaited or returned  jest/valid-expect
  614:9  error  Async assertions must be awaited or returned  jest/valid-expect
  615:9  error  Async assertions must be awaited or returned  jest/valid-expect
```
I think those might just be edge cases suitable for `eslint-disable` though, you can't cover every form of waiting for promises anyway. @SimenB

Reply:
Regarding my comment about `parent`s:

As far as `expectCase` is concerned, I think you're likely right that it won't break as long as it's called with `CallExpression`s as a node because those will always have at least an `ExpressionStatement` and a `Program` above them. I'd personally still prefer to keep the parents check in because as a rule author I'd expect `expectCase` to be safe (never throw regardless of what node I give it). Removing the `arguments.length` check sounds good to me!

As far as `getParentCallExpression` is concerned, here's an example that crashes the rule: `expect().resolves.asdf;`
(A weird line of course, but still linting should never crash, only report errors)

Reply:
@jeysal thank you very much for the feedbacks.

1. I noticed that it's possible to use `.not`, after `.resolves`. I'll make sure that it doesn't break anything.
  https://github.com/facebook/jest/blob/db5e472b51cfc68787c9da7bb21b26f192b0ab78/packages/expect/src/__tests__/matchers.test.js#L1179

2. Regarding having the assertion inside a Promise method, for `Promise.fn(expect())` or `Promise.fn([expect()])` usages, I can also check if Promise was awaited or returned. I think it's straightforward to solve for this case and we can save devs from some pain?

3. I'll look into expectCase function and add the necessary checks. I didn't notice it was crashing. 

Also, is there already a setup to test the plugin on an existing project? How did you run it on jest for instance? Is there some instructions to setup that? Thanks.

Reply:
`Promise.all` & `Promise.race`, and the new `Promise.allSettled` & `Promise.any` should all be supported and count as a new promise, unless implementing that is super complicated

Reply:
@SimenB it's not complicated at all. I already did that just couldn't push it yet. I simply check for all `Promise.x` usages where x can be any of those methods. I will push it this weekend.

Reply:
Sounds perfect 🙂

Reply:
> Also, is there already a setup to test the plugin on an existing project? How did you run it on jest for instance?

1. `yarn link` in the `eslint-plugin-jest` dir
2. `yarn link eslint-plugin-jest` in the `jest` dir
3. `yarn lint` in the `jest` dir

:)